### ISSUE TITLE
Remove the Bag-of-Words block: retrain with --deploy C (more accurate AND safer)

### ISSUE THREAD

**Issue by Muawiya-contact** [Muawiya-contact/ML_Predictor#30]
## Short version

The deployed model carries a Bag-of-Words block it does not need. Removing it
makes the model **more accurate and safer at the same time**, and cuts the
feature count by almost two thirds.

Removing it means retraining once with a different flag. Nothing else changes.

---

## What is in the model right now

The deployed bundle is `triage_model_embedding_english/`. Every prediction is
built from a 1,110-number feature vector, made of three parts:

| Part | Size | What it is |
|---|---|---|
| Structured | 26 | age, heart rate, blood pressure, temperature, SpO2, plus gender / arrival / AVPU / ECG as one-hot |
| **Bag of Words** | **700** | **350 word counts + 350 character counts, each multiplied by a hand-written weight** |
| Sentence embedding | 384 | the multilingual sentence encoder |

This is configuration **D (hybrid)** in `train_embedding_pipeline.py`.

## Why the BoW block should go

When the four text representations were compared during training, the
configuration **without** the BoW block (**C**) beat the one with it (**D**) on
every single measure. Both numbers come from the same run, on the same 451
held-out test cases, so this is a fair comparison — not an estimate.

| Measure | D (what we ship) | C (no BoW) | Change |
|---|---|---|---|
| Accuracy | 76.05% | **80.49%** | **+4.44** |
| Under-triage (dangerous direction) | 13.08% | **12.64%** | **−0.44** |
| Over-triage | 10.86% | **6.87%** | **−3.99** |
| Feature count | 1,110 | **410** | **−700** |

There is no measure where D wins. It is more accurate to drop the block, safer
in both error directions, and the model becomes 63% smaller.

## Why we kept it, and why that reason turned out to be wrong

The reason for keeping the BoW block was **explainability**. The idea was that
because BoW features have real names ("chest", "dard"), the model could show a
clinician which words pushed a complaint to a particular triage level. An
embedding cannot do that, so the accuracy loss looked like a fair trade.

**That feature was never actually built.** Checked the whole codebase:

- Nothing anywhere reads `model.coef_`.
- Nothing attributes a prediction to individual words.
- `feature_names` and `attention` are used in exactly one place —
  `triage_pipeline.py` lines 1130-1136 and 1230-1233 — and only to build the
  BoW block itself. Nothing consumes the result.

The GUI *does* have explanation views, but they all work on **text stages**,
not on model features:

- Pipeline Explorer shows the normalised text, the translation, and which
  tokens stop-word removal dropped.
- The Stop Words tab explains why each word was kept or removed.
- The safety gate explains which body part it expected and what it found.

Every one of those keeps working exactly the same with the BoW block gone. They
do not touch the classifier's coefficients at all.

So we are paying 4.44 points of accuracy for a benefit that does not exist in
the code.

## One more thing worth knowing

The training script's own default is already `--deploy C`:

```python
parser.add_argument("--deploy", default="C", choices=["auto", "A", "B", "C", "D"], ...)
```

Shipping D took an explicit override of that default. So this is really about
going back to what the script already considers the normal choice.

---

## How to fix it

One command:

```bash
python train_embedding_pipeline.py --deploy C
```

Then check the new `triage_model_embedding_english/triage_metrics.json` shows
roughly:

- accuracy ≈ **80.49%**
- under-triage ≈ **12.64%**
- over-triage ≈ **6.87%**
- `n_features` = **410**
- `text_representation` = `embeddings_preprocessed`

The seed is fixed (`random_state=42`), so it should reproduce those numbers
closely. **If it does not, stop and investigate before shipping the bundle** —
a big difference would mean something else changed.

## Then check nothing broke

```bash
python tests/audit_pipeline.py
python tests/audit_gui.py
```

Both should still pass. The serving code already handles a bundle with no BoW
block — `build_text_features()` loops over whatever blocks the manifest lists,
so a C bundle just has one fewer block to build. `load_artifacts()` only loads
`word_bow.pkl` / `char_bow.pkl` when `'bow' in blocks`. **No code change should
be needed.** If a code change turns out to be needed, that is worth noting on
this issue.

Also worth a quick manual check afterwards:

- Run one prediction in the GUI and confirm a level still comes out.
- Run the batch fixture (`tests/fixtures/batch_sample.csv`) and confirm the CSV
  still exports all its columns.

## Cleanup that becomes possible afterwards

Once no shipped bundle uses BoW, this becomes dead code and could be removed in
a later pass (does not have to happen at the same time):

- `MEDICAL_WEIGHTS` — the 120-key hand-written weight table in
  `triage_pipeline.py`
- `build_attention_weights()` and `apply_attention()`
- The `'bow'` branch inside `build_text_features()`

Leave it alone for now if in doubt — configurations A and D still use it, and
the ablation in the paper still reports them.

## Documentation to update

- `docs/methodology.html` → re-render `docs/ML_Predictor_Methodology.pdf`.
  §3.9.1 already documents this finding and has a table for it, but the results
  section (§3.10, Figure 7, Table 4) is still D's numbers and would need the new
  confusion matrix.
- `model_manifest.json` will be rewritten by training automatically.
- Check `README.md` and `SUBMISSION_SUMMARY.md` for accuracy figures that would
  go stale.

---

## Careful

The **2,252-row problem does not go away**. Under-triage improves to 12.64%,
which is still above the <10% target and still worse than the 9.25% of the older
10,000-row Roman Urdu model. This change is a real improvement, but it is not a
fix for the small training set.

Also: this replaces the currently submitted artifact. If the current numbers
have already been reported to anyone, decide whether to redo this before or
after that matters.
